pinot-controller/src/main/java/org/apache/pinot/controller/api/access/AccessControlUtils.java

```diff
@@ -43,24 +44,25 @@ private AccessControlUtils() {
   /**
    * Validate permission for the given access type against the given table
    *
-   * @param tableName name of the table to be accessed (post database name translation)
-   * @param accessType type of the access
-   * @param httpHeaders HTTP headers containing requester identity required by access control object
-   * @param endpointUrl the request url for which this access control is called
+   * @param tableName     name of the table to be accessed (post database name translation)
+   * @param accessType    type of the access
+   * @param httpHeaders   HTTP headers containing requester identity required by access control object
+   * @param request       the request for which this access controll is called
+   * @param endpointUrl   the request url for which this access control is called
    * @param accessControl AccessControl object which does the actual validation
    */
   public static void validatePermission(@Nullable String tableName, AccessType accessType,
-      @Nullable HttpHeaders httpHeaders, String endpointUrl, AccessControl accessControl) {
+      @Nullable HttpHeaders httpHeaders, Request request, String endpointUrl, AccessControl accessControl) {
```

Comment:
Can you create a new method with the `request` param and deprecate the existing one, instead of changing the existing method?

Reply:
That's correct.